src/app/modules/OpenshiftStreams/dialogs/CreateInstance/api.ts

```diff
@@ -204,6 +204,10 @@ export const useCreateInstance = (): OnCreateKafka => {
           case ErrorCodes.INSUFFICIENT_QUOTA:
             onError('over-quota');
             break;
+          case ErrorCodes.REACHED_MAX_LIMIT_ALLOWED_KAFKA:
```

Comment:
From @riccardo-forina: 
> I know error 24 can happen for quota instances too, but can it actually? If that's the case I think we need to expand the errors handled by the UI to cover for this case , or generalize the trial-unavailable we are using now to remove the reference to the trial. As seen in the screenshot e don't write trial in the error message, but the string in the code does and it's confusing.

If you're referring to [this error 24](https://github.com/redhat-developer/app-services-sdk-js/blob/main/packages/kafka-management-sdk/src/errors.ts#L55), could someone clarify what cases would result in that error?
```
  /** The maximum number of allowed kafka instances has been reached*/
  ERROR_24 : "KAFKAS-MGMT-24", 
```

The use cases that we provided designs for are:
1. Trial user - user already has an instance
    > You already have an instance running
    > You can deploy 1 instance at a time. To deploy a new instance, delete your existing one first.
    - Do we know what error code(s) we get for this use case? There's also the following error code, and it isn't clear when these different error codes are returned (e.g. on form load, on form submit?)
    ```
      /** Forbidden to create more instances than the maximum allowed*/
      ERROR_5 : "KAFKAS-MGMT-5", 
    ```
2. User with subscription - org has reached quota
    > Kafka instances are unavailable for creation
    > Your organization reached the limit of Kafka instances available for creation.
3. Trial user - This wasn't a scenario that came up when we were working on designs. If there is also a max number allowed for orgs without a subscription, then I would expect the same message shown for 2 to display for the trial user, too.

Reply:
Error 24 happens because backend have limit of maximum instances we can create per region and per instance type.

Use cases are really good and it will be still pretty much valid for the new designs that have kafka sizes/streaming units. It will be good to copy this to some space to not lose it in on github

Reply:
Ah, ok. Thanks for the clarification. So for that error code, we did cover the use case for the trial user, but not really for the subscribed user. This use case for the subscribed user is something I want to address in the redesign of the modal. 

To see the use cases we currently support in the modal, they're all listed in our storybook under *Create Kafka Instance*: https://main--619e1921b3e38b003afeb2be.chromatic.com/

A nice enhancement to these stories is to note what error codes map to which examples. I'm struggling to keep that information in my head. 🤪

Reply:
Back to @riccardo-forina's original comment:
> I know error 24 can happen for quota instances too, but can it actually? If that's the case I think we need to expand the errors handled by the UI to cover for this case , or generalize the trial-unavailable we are using now to remove the reference to the trial. As seen in the screenshot e don't write trial in the error message, but the string in the code does and it's confusing.

The messages we provided in the last design *are* specific to the trial user. I would not expect this error to be likely given the limited number of customers and we're actively watching available regions. But I do want to provide examples for how to handle this case in the next design of this modal. 

And we don't explicitly state "trial" or "evaluation" in the current design. There was probably a reason for that, which I don't remember. The main way to see from the UI that it is creating a trial instance is with the "Duration: 48 hours" text that displays on the right. 

I think it's fine to do what you're suggesting as part of the next set of updates, if that helps to keep the updates in this PR simple for the issue we're trying to fix. But I can easily be convinced otherwise.